PDF invoice generation

Lenetis opened the pull request:
Example invoice: https://bstrzelecki25.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-12

Lenetis committed:
Working pdf generation + invoice template and css

Lenetis committed:
Redesigned XML invoice generation, added Invoice class, added XML template

Lenetis committed:
Fixed imports and tests, reordered requirements

Lenetis committed:
Using the new Invoice class in tests

Lenetis committed:
Updated pdf template

Lenetis committed:
Removed out.pdf file

Lenetis committed:
Fixed indentation

Lenetis committed:
Changed extension of Jinja templates

Lenetis committed:
Changed type of client packages

Lenetis committed:
Fixed invoice css

bstrzelecki commented on test/invoice_generator_test.py:
missing test case for saving pdf

Lenetis replied:
added test_generate_pdf()

bstrzelecki commented on src/invoice_generator.py:
```diff
@@ -1,54 +1,112 @@
-import xml.etree.cElementTree as ET
-from typing import Dict
+from dataclasses import dataclass, asdict
+from typing import Dict, List, Tuple
```
any reason for using deprecated types?

Lenetis replied:
I didn't know they were deprecated. Changed to default dict, list, tuple

Lenetis committed:
Removed from typing import Dict, List,  Tuple

Lenetis committed:
Added pdf test, improved xml test

Lenetis committed:
Fix test failing on github + little improvements

Lenetis committed:
Added test for the deprecated function because github test coverage complained

Lenetis committed:
Fix

Lenetis committed:
Added install dependencies workflow

Lenetis committed:
Enable pdf test

Lenetis committed:
Moved wkhtmltopdf installation to main checkout workflow

Lenetis committed:
Fixed pdf test failing on Linux

(On Windows pdfkit separates html tables with spaces, on Linux with tabs. Fixed by splitting text by any whitespace)

Lenetis committed:
Merge branch 'main' into invoice-generation-pdf

bstrzelecki commented on src/invoice_generator.py:
```diff
@@ -1,54 +1,111 @@
+    total_cost: float
+    invoice_date: datetime.date
+    payment_due: datetime.date
+    company_name: str = "Drying Paint Inc."
```
there is already merged pull request containing file that included all necessary constants (manifest.py)

bstrzelecki reviewed (commented):
DeprecationWarning: PyPDF2 is deprecated. Please move to the pypdf library instead.

Lenetis committed:
Added data from manifest file

Lenetis committed:
Merge branch 'main' into invoice-generation-pdf

Lenetis committed:
Switched back to PyPDF2
